src/main/java/BotChat/Event.java

```diff
@@ -0,0 +1,81 @@
+package BotChat;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents an event task with a start and end date/time in the botChat application.
+ * Extends the base Task class.
+ */
+public class Event extends Task {
+    protected String from;
+    protected String to;
+
+    /**
+     * Constructs an Event object with the specified description, start date/time, and end date/time.
+     * If the provided dates are in a valid format, they are used as-is; otherwise, they are converted.
+     *
+     * @param description The description of the event task.
+     * @param from        The starting date/time of the event in its original format.
+     * @param to          The ending date/time of the event in its original format.
+     */
+    public Event(String description, String from, String to) {
+        super(description);
+        if (isValidDateFormat(from) && isValidDateFormat(to)) {
+            this.from = from;
+            this.to = to;
+        } else {
+            this.from = convertDate(from);
+            this.to = convertDate(to);
+        }
+    }
+    public String getFrom() {
+        return from;
+    }
+    public String getTo() {
+        return to;
+    }
+
+    /**
+     * Checks if the provided date is in a valid format.
+     *
+     * @param by The date to be checked.
+     * @return True if the date is in a valid format, false otherwise.
+     */
+    private boolean isValidDateFormat(String by) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy");
+        try {
+            LocalDate date = LocalDate.parse(by, formatter);
+            String formattedDateTime = date.format(formatter);
+            return formattedDateTime.equals(by);
+        } catch (Exception e) {
+            return false;
+        }
+    }
```

Comment:
As an example for the review regarding a DateTime class in `TaskList`, these could be left for the Date time class to handle. It would help to clean up the code with less repetition. Well done overall!